src/ex1/scripts/domScripts.js

```diff
@@ -0,0 +1,195 @@
+
+taskInput.addEventListener("keypress", function (event) {
+  if (event.key === "Enter") {
+    event.preventDefault();
+    addTaskButton.click();
+  }
+});
+
+function onAddTaskClick(e) {
+  const isOk = canProceed();
+  if (isOk) {
+    const taskDiv = document.createElement("div");
+    taskDiv.classList.add("task");
+
+    const gripLines = createGridLines();
+    taskDiv.appendChild(gripLines);
+
+    const taskContent = createTaskContent();
+    taskDiv.appendChild(taskContent);
+
+    const completeTaskButton = createCompleteBtn();
+    taskDiv.appendChild(completeTaskButton);
+
+    const removeTaskButton = createRemoveBtn();
+    taskDiv.appendChild(removeTaskButton);
+
+    addTaskDivEventListeners(taskDiv);
+    tasksList.appendChild(taskDiv);
+    saveTaskToLocalStorage(taskContent.innerText, false);
+    toggleEmptyMsg();
+    taskInput.value = "";
+  }
+}
+
+function canProceed() {
+  const isAlertOn = isAlertShown();
+  if (taskInput.value === "") {
+    if (!isAlertOn) {
+      toggleAlert();
+    }
+    return false;
+  } else {
+    if (isAlertOn) {
+      toggleAlert();
+    }
+    return true;
+  }
+}
+
+function createGridLines() {
+  const gripLines = document.createElement("div");
+  gripLines.classList.add("grip-lines-icon");
+  gripLines.innerHTML = '<i class="fas fa-grip-lines"></i>';
+  return gripLines;
+}
+
+function createTaskContent(content = null) {
+  const newTask = document.createElement("p");
+  newTask.classList.add("task-item");
+  if (content) {
+    newTask.innerText = content;
+  } else {
+    newTask.innerText = taskInput.value;
+  }
+  return newTask;
+}
+
+function createCompleteBtn(isCompleted = false) {
+  const completeTaskButton = document.createElement("button");
+  completeTaskButton.classList.add("task-complete-btn");
+  completeTaskButton.classList.add("hide");
+  if (isCompleted) {
+    completeTaskButton.classList.add("btn-completed");
+  }
+  completeTaskButton.innerHTML = '<i class="fas fa-check"></i>';
+  return completeTaskButton;
+}
+
+function createRemoveBtn() {
+  const removeTaskButton = document.createElement("button");
+  removeTaskButton.classList.add("task-remove-btn");
+  removeTaskButton.classList.add("hide");
+  removeTaskButton.innerHTML = '<i class="fas fa-trash"></i>';
+  return removeTaskButton;
+}
+
+function addTaskDivEventListeners(taskDiv) {
+  taskDiv.addEventListener("click", onTaskClick);
+  taskDiv.addEventListener("mouseover", onTaskMouseOver);
+  taskDiv.addEventListener("mouseleave", onTaskMouseOut);
+  taskDiv.draggable = true;
+}
+
+function onTaskButtonsClick(e) {
+  const button = e.target;
+  if (button.classList.contains("task-complete-btn")) {
+    button.parentElement.classList.toggle("task-completed");
+    button.classList.toggle("btn-completed");
+    updateTaskInLocalStorage(
+      button.parentElement.querySelector(".task-item").innerText
+    );
+  } else if (e.target.classList.contains("task-remove-btn")) {
```

Comment:
it is a bad practice to create a general event listener and then check inside the code which class the element hold in order to do a certain functionality.

A better way  will be to do what you did many times before - attach a specific click event to the button element when you created the element at first place (either complete or remove)